File: Java Script - Introduction to Programming/6.2. Drawing on the Console - Exam Problems/02. Butterfly.js
function butterfly(n) {
  //top part
  for (i = 0; i < n - 2; i++) {
    if (i % 2 === 0) {
      console.log('*'.repeat(n - 2) + '\\' + ' ' + '/' + '*'.repeat(n - 2))
    } else {
      console.log('-'.repeat(n - 2) + '\\' + ' ' + '/' + '-'.repeat(n - 2))
    }
  }
  //middle part
  console.log(' '.repeat(n - 1) + '@')

  //bottom part
  for (i = 0; i < n - 2; i++) {
    if (i % 2 === 0) {
      console.log('*'.repeat(n - 2) + '/' + ' ' + '\\' + '*'.repeat(n - 2))
    } else {
      console.log('-'.repeat(n - 2) + '/' + ' ' + '\\' + '-'.repeat(n - 2))
    }
  }
}

// butterfly(7)

File: Pyrhon - Introduction to Programming/6.2. Drawing on the Console - Exam Problems/02. Butterfly.py
n = int(input())
#top part
for i in range(n - 2):
    if i & 1:
      print('-' * (n - 2) + '\\' + ' ' + '/' + '-' * (n - 2))
    else:
      print('*' * (n - 2) + '\\' + ' ' + '/' + '*' * (n - 2))

#middle part
print(' ' * (n - 1) + '@')

#bottom part
for i in range(n - 2):
  if i & 1:
    print('-' * (n - 2) + '/' + ' ' + '\\' + '-' * (n - 2))
  else:
    print('*' * (n - 2) + '/' + ' ' + '\\' + '*' * (n - 2))
